File: aulas/20160428/01-Static/Static.java
class MyType<T>
{
    private static int val;
    
    public int  getVal()          { return val;  }
    public void setVal(int value) { val = value; }
}

public class Static
{
    public static void main(String[] args)
    {
        MyType<Integer> mt_int = new MyType<Integer>();
        MyType<String>  mt_str = new MyType<String>();
        
        mt_int.setVal(3);
        mt_str.setVal(7);
        
        System.out.println(mt_int.getVal());
        System.out.println(mt_str.getVal());
    }
}

File: aulas/20160428/01-Static/Static.cs
using System;

class MyType<T>
{
    private static int val;
    
    public int Val
    {
        get { return val;  }
        set { val = value; }
    }
}

public class Static
{
    public static void Main()
    {
        MyType<int>    mt_int = new MyType<int>();
        MyType<string> mt_str = new MyType<string>();
        
        mt_int.Val = 3;
        mt_str.Val = 7;
        
        Console.WriteLine(mt_int.Val);
        Console.WriteLine(mt_str.Val);
    }
}
